File: node_server/models/course.js

module.exports = (sequelize, DataTypes) => {
  const Course = sequelize.define('Course', {
    name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
  }, {});

  Course.associate = (models) => {
    Course.belongsToMany(models.User, {
      as: 'instructors',
      through: 'UserCourse',
    });
    Course.belongsToMany(models.User, {
      as: 'students',
      through: 'UserCourse',
    });
  };

  return Course;
};

File: server/model/course.go
package model

import (
	"github.com/jinzhu/gorm"
)

type Course struct {
	gorm.Model
	Name        string `gorm:"NOT NULL"`
	Instructors []User `gorm:"many2many:instructor_courses`
	Students    []User `gorm:"many2many:student_courses`
}
